### ISSUE TITLE
Enhancements to subtitle list mode

### ISSUE THREAD

**Issue by helica1** [lrorpilla/jidoujisho#198]
First of all, thank you so much for this amazing program.

I love the new enhancements to subtitle list mode, where we can now
play audio from the subtitle list. I have now spent several hours
reading along with the audio for audiobooks synced to the transcripts
and think it's a really good supplement to normal reading. The
implementation of it is nearly perfect and it makes a very smooth
reading experience.

I do have a few feature requests to consider:
1) "Immersion" (distraction-free) mode -> where all subtitle timings
are hidden. Ideally this would also be a full-screen mode where the
clock/status bar are hidden. Maybe it's my own fault but I find it a
little hard to concentrate when I'm reminded of my progress with every
line. It would be ideal to even hide everything except the text: hide
the buttons to the right, hide the subtitle timings, and hide the
system bar. A toggle switch at the top of the screen to turn immersion
mode on and off would be great.

2) Audio "padding" -> especially when using bluetooth headphones it
sometimes takes a few seconds for the sound to start again, resulting
in a few seconds of missed audio. This is just an android issue, but
to compensate it would be nice to be able to specify a few seconds (5
seconds? 10 seconds?) of padding, so that when playback is resumed
after being paused, the playhead automatically rewinds by the
specified time.

3) Search for subtitle. It would be really nice to be able to search
for a playback location by searching for a string in a subtitle. Read
the book normally in reader mode from epub -> Search for a sentence in
the audiobook/subtitle list mode and bam! -> continue from that spot
by listening to the audiobook in subtitle list mode. I imagine this
might be a little harder to implement than the other two feature
requests about but would be nice.

Thank you again for your hard work and for considering these enhancements!

**Comment by lrorpilla** [comment]
1. The status bar being there actually has a purpose. If it's not there, using a back gesture will show the status bar instead of executing back immediately. It would require two motions.

2. I would just recommend selecting the previous item instead of exactly the item you want.

3. This is a bit complicated to add (not technically hard, just a lot of work), but it's a reasonable add for the use case.

I think these suggestions aren't ones that get me to rush to my seat as soon as possible, and are more "nice to haves", so I'll leave this one open.

Just as expectation to future requests -- I'm going on hiatus now and I really mean it this time, I'm a bit busy and have my hands full with work now and I'll be doing critical bug requests if needed but otherwise the app will be as is in the meantime, as a heads-up.

**Comment by helica1** [comment]
Thanks for considering.

For #1 - keeping the status bar is fine. Agree it is best to minimize the
gestures. But what about hiding the timestamps listed with every subtitle?

Thanks again for carefully considering everything - very impressive how you
are able to accomodate so many requests while also keeping everything so
streamlined with a clear vision.

**Comment by lrorpilla** [comment]
All three requests are implemented in [**2.6.6**](https://github.com/lrorpilla/jidoujisho/releases/tag/2.6.6), so I'm closing this issue. Please let me know if you have any concerns or feedback with the implementation.

**Comment by lrorpilla** [comment]
Reuploaded just now with decreased left padding for subtitles in the transcript.

**Comment by helica1** [comment]
Thank you so much! It is surprising what a positive difference it makes to
hide the subtitle timings - it really is more like reading a book and less
like reading a transcript.

Also, the search function works perfectly - it is a game-changer to be
reading a book, search for a phrase and continue in the audiobook in the
exact position.

Just a small thing but it doesn't seem like the audio padding is working.
Tested with bluetooth headphones and built-in speaker (Galaxy Fold 3), and
when pausing and resuming there are still a few seconds of skipped audio.
Relatively minor - as you said I can click the previous subtitle. However
in this "book mode" several subtitle chunks are nearly a paragraph long so
that's not always ideal. In a perfect would would be nice to rewind a few
seconds to repeat when was just said (as this is usually the new word/etc).

Thanks again!

**Comment by lrorpilla** [comment]
There should be a moderate improvement on the audio delays, but I acknowledge it's not completely fixed just yet.

I think I am determined to try and completely fix the audio issue instead of band-aiding it, which involves studying some issues involving VLC Android's implementation. 

This issue would unblock #42 or #93, so I am determined to solve this problem outright instead of work around it. I encourage you to make a new issue to track the audio lag problem separate from this issue.

**Comment by lrorpilla** [comment]
@helica1, can you try [**2.6.7**](https://github.com/lrorpilla/jidoujisho/releases/tag/2.6.7) and tell me how the audio delay is for you?

**Comment by helica1** [comment]
Thank you for fixing this so quickly. I can confirm that 2.6.7 fixes both
the audio delay issue, and also the subtitle display error when toggling
timestamps on/off. Both work exactly as expected now, thank you!

**Comment by lrorpilla** [comment]
That's great to hear. Enjoy!
